### ISSUE TITLE
ZIM files not detected on a BB10 (Android Jelly bean)

### ISSUE THREAD

**Issue by kelson42** [kiwix/kiwix-android#124]
Kiwix v2.2 and also previous versions cannot locate the ZIM files in my device (i tried storing in different paths) however from a file explorer I can click on a ZIM file and Kiwix will open the file.

The device is a BB10 using an Android Jelly bean runtime and the path mapping between internal QNX filesystem and runtime is something like :

Internal mem:
* /mnt/sdcard/ (Android runtime)
* /sdcard (alias to /mnt/sdcard/)
* /misc/android/
* /accounts/1000/shared/misc/android (BB filesystem):

Android folder in internal mem
* /mnt/sdcard/Android (Android runtime)
* /misc/android/android (alias to /mnt/sdcard/misc/android/android)
* /accounts/1000/shared/misc/android (BB filesystem):

external/SD card memory:
* /mnt/sdcard/external_sd/ (Android runtime)
* /accounts/1000/removable/sdcard/ (BB filesystem):

Bug firstly reported here https://sourceforge.net/p/kiwix/feature-requests/1008

**Comment by kelson42** [comment]
@mhutti1 Any idea?

**Comment by dapgo** [comment]
Thanks for create the ticket, 

These ZIMs files are stored in:
/mnt/sdcard/kiwix (for bb10 native apps path seems mapped to /misc/android/kiwix)
/mnt/sdcard/external_sd/kiwix_sd

Being able to specify a path i could try and figure out where is the problem. In other android apps i have no problems to find files, but with few other opensource apps such as "aard" i had the same problem.

**Comment by mhutti1** [comment]
All those paths should be checked except /misc/android/kiwix and /accounts/1000/. We use two methods to search for files so it is interesting that neither would work. I can try add those 2 paths to an apk to test on a device if you think it might work?

**Comment by dapgo** [comment]
if you can add these paths, it would be great and I'd happily test it. ;)

**Comment by mhutti1** [comment]
@dapgo sorry it took so long. I have exams at the moment. Also sorry its so big, we have a small problem with our libs at the moment. [link](https://drive.google.com/open?id=0Bwjmzre_HGWzSHFoTHdmRTdsb0E)

**Comment by dapgo** [comment]
No sorry, 
I tried it with success because  now Kiwix can detect ZIM files :) 
For your information in the bottom of the page it seems that kiwix is still looking for dictionaries forever, but it doesn't freeze the app so it is possible to select a dictionary and work.

![kiwix_zim_detected_bb10](https://cloud.githubusercontent.com/assets/4981669/25798780/c9bc0804-33e2-11e7-8fab-8582255811d3.png)

**Comment by kelson42** [comment]
@dapgo Good news :)
@mhutti1 Might that be that we do not detect recursive infinite loops in fs scanning?

**Comment by dapgo** [comment]
logging the output:

I/QNXNavigatorClient(13082844): canJoinGroup: cid=1964612583842017 (org.kiwix.kiwixmobile) shell group already joined
I/Zygote  (13082844): Process: zygote socket opened
I/ActivityManager(13082844): Start proc com.svox.pico for service com.svox.pico/.PicoService: pid=53616884 uid=10136 gids={50136, 1015, 1028}
I/TextToSpeech(53502164): Sucessfully bound to com.svox.pico
E/kiwix   (53502164): Unable to open the ZIM fulltext index /mnt/sdcard/kiwix/wikiquote_en_all_nopic_2017-03.zim
I/SVOX Pico Engine(53616884): loaded en-GB successfully
I/dalvikvm(53502164): Could not find method android.text.Html.fromHtml, referenced from method org.kiwix.kiwixmobile.utils.StyleUtils.fromHtml
W/dalvikvm(53502164): VFY: unable to resolve static method 21710: Landroid/text/Html;.fromHtml (Ljava/lang/String;I)Landroid/text/Spanned;
I/dalvikvm(53502164): Could not find method org.kiwix.kiwixmobile.KiwixMobileActivity.getExternalMediaDirs, referenced from method org.kiwix.kiwixmobile.views.web.KiwixWebView$1.handleMessage
W/dalvikvm(53502164): VFY: unable to resolve virtual method 48686: Lorg/kiwix/kiwixmobile/KiwixMobileActivity;.getExternalMediaDirs ()[Ljava/io/File;
I/kiwix   (53502164): Intent data: null
I/TextToSpeech(53502164): Connected to ComponentInfo{com.svox.pico/com.svox.pico.PicoService}
I/TextToSpeech(53502164): Set up connection to ComponentInfo{com.svox.pico/com.svox.pico.PicoService}
I/Choreographer(53502164): Skipped 51 frames!  The application may be doing too much work on its main thread.
I/ActivityManager(13082844): Displayed org.kiwix.kiwixmobile/.KiwixMobileActivity: +2s972ms
E/Web Console(53502164): Uncaught TypeError: Cannot call method 'start' of undefined at null:1
E/Web Console(53502164): Uncaught TypeError: Cannot call method 'start' of undefined at null:1
W/dalvikvm(53502164): method Landroid/support/v7/widget/ListViewCompat;.lookForSelectablePosition incorrectly overrides package-private method with same name in Landroid/widget/ListView;
E/dalvikvm(53502164): Could not find class 'android.transition.Transition', referenced from method android.support.v7.widget.MenuPopupWindow.setEnterTransition
W/dalvikvm(53502164): VFY: unable to resolve check-cast 2685 (Landroid/transition/Transition;) in Landroid/support/v7/widget/MenuPopupWindow;
E/dalvikvm(53502164): Could not find class 'android.transition.Transition', referenced from method android.support.v7.widget.MenuPopupWindow.setExitTransition
W/dalvikvm(53502164): VFY: unable to resolve check-cast 2685 (Landroid/transition/Transition;) in Landroid/support/v7/widget/MenuPopupWindow;
I/ListPopupWindow(53502164): Could not find method setEpicenterBounds(Rect) on PopupWindow. Oh well.
I/dalvikvm(53502164): Could not find method android.widget.PopupWindow.showAsDropDown, referenced from method android.support.v7.widget.AppCompatPopupWindow.showAsDropDown
W/dalvikvm(53502164): VFY: unable to resolve virtual method 23323: Landroid/widget/PopupWindow;.showAsDropDown (Landroid/view/View;III)V
I/dalvikvm(53502164): Could not find method android.support.v7.widget.DropDownListView.drawableHotspotChanged, referenced from method android.support.v7.widget.DropDownListView.setPressedItem
W/dalvikvm(53502164): VFY: unable to resolve virtual method 18744: Landroid/support/v7/widget/DropDownListView;.drawableHotspotChanged (FF)V
I/dalvikvm(53502164): Could not find method android.view.View.drawableHotspotChanged, referenced from method android.support.v7.widget.DropDownListView.setPressedItem
W/dalvikvm(53502164): VFY: unable to resolve virtual method 22138: Landroid/view/View;.drawableHotspotChanged (FF)V
I/Choreographer(53502164): Skipped 51 frames!  The application may be doing too much work on its main thread.

**Comment by dapgo** [comment]
Loggin the output:

I/QNXNavigatorClient(13082844): canJoinGroup: cid=1964612583842017 (org.kiwix.kiwixmobile) shell group already joined
I/Zygote  (13082844): Process: zygote socket opened
I/ActivityManager(13082844): Start proc com.svox.pico for service com.svox.pico/.PicoService: pid=53616884 uid=10136 gids={50136, 1015, 1028}
I/TextToSpeech(53502164): Sucessfully bound to com.svox.pico
E/kiwix   (53502164): Unable to open the ZIM fulltext index /mnt/sdcard/kiwix/wikiquote_en_all_nopic_2017-03.zim
I/SVOX Pico Engine(53616884): loaded en-GB successfully
I/dalvikvm(53502164): Could not find method android.text.Html.fromHtml, referenced from method org.kiwix.kiwixmobile.utils.StyleUtils.fromHtml
W/dalvikvm(53502164): VFY: unable to resolve static method 21710: Landroid/text/Html;.fromHtml (Ljava/lang/String;I)Landroid/text/Spanned;
I/dalvikvm(53502164): Could not find method org.kiwix.kiwixmobile.KiwixMobileActivity.getExternalMediaDirs, referenced from method org.kiwix.kiwixmobile.views.web.KiwixWebView$1.handleMessage
W/dalvikvm(53502164): VFY: unable to resolve virtual method 48686: Lorg/kiwix/kiwixmobile/KiwixMobileActivity;.getExternalMediaDirs ()[Ljava/io/File;
I/kiwix   (53502164): Intent data: null
I/TextToSpeech(53502164): Connected to ComponentInfo{com.svox.pico/com.svox.pico.PicoService}
I/TextToSpeech(53502164): Set up connection to ComponentInfo{com.svox.pico/com.svox.pico.PicoService}
I/Choreographer(53502164): Skipped 51 frames!  The application may be doing too much work on its main thread.
I/ActivityManager(13082844): Displayed org.kiwix.kiwixmobile/.KiwixMobileActivity: +2s972ms
E/Web Console(53502164): Uncaught TypeError: Cannot call method 'start' of undefined at null:1
E/Web Console(53502164): Uncaught TypeError: Cannot call method 'start' of undefined at null:1
W/dalvikvm(53502164): method Landroid/support/v7/widget/ListViewCompat;.lookForSelectablePosition incorrectly overrides package-private method with same name in Landroid/widget/ListView;
E/dalvikvm(53502164): Could not find class 'android.transition.Transition', referenced from method android.support.v7.widget.MenuPopupWindow.setEnterTransition
W/dalvikvm(53502164): VFY: unable to resolve check-cast 2685 (Landroid/transition/Transition;) in Landroid/support/v7/widget/MenuPopupWindow;
E/dalvikvm(53502164): Could not find class 'android.transition.Transition', referenced from method android.support.v7.widget.MenuPopupWindow.setExitTransition
W/dalvikvm(53502164): VFY: unable to resolve check-cast 2685 (Landroid/transition/Transition;) in Landroid/support/v7/widget/MenuPopupWindow;
I/ListPopupWindow(53502164): Could not find method setEpicenterBounds(Rect) on PopupWindow. Oh well.
I/dalvikvm(53502164): Could not find method android.widget.PopupWindow.showAsDropDown, referenced from method android.support.v7.widget.AppCompatPopupWindow.showAsDropDown
W/dalvikvm(53502164): VFY: unable to resolve virtual method 23323: Landroid/widget/PopupWindow;.showAsDropDown (Landroid/view/View;III)V
I/dalvikvm(53502164): Could not find method android.support.v7.widget.DropDownListView.drawableHotspotChanged, referenced from method android.support.v7.widget.DropDownListView.setPressedItem
W/dalvikvm(53502164): VFY: unable to resolve virtual method 18744: Landroid/support/v7/widget/DropDownListView;.drawableHotspotChanged (FF)V
I/dalvikvm(53502164): Could not find method android.view.View.drawableHotspotChanged, referenced from method android.support.v7.widget.DropDownListView.setPressedItem
W/dalvikvm(53502164): VFY: unable to resolve virtual method 22138: Landroid/view/View;.drawableHotspotChanged (FF)V
I/Choreographer(53502164): Skipped 51 frames!  The application may be doing too much work on its main thread.

**Comment by mhutti1** [comment]
Possible but unlikely. It could just take a long time to scan.

**Comment by kelson42** [comment]
@mhutti1 If the fix is merge, on my side we can close that ticket.

**Comment by mhutti1** [comment]
Haven't merged yet. I will do soon.

**Comment by kelson42** [comment]
@mhutti1 Does that have been done meanwhile?

**Comment by mhutti1** [comment]
@kelson42 Its not been done yet. As I said its 2 lines. I will do it in a few days time hopefully.

**Comment by dapgo** [comment]
BTW, using the the previous modified apk , I detected that after installation and scan, zim files are detected (in internal an external memory).
However,  after a while I decided to free internal mem so  I moved some files from internal memory to external memory, but the scan never end and files were not detected, I tried several times and waiting for a very long time.

So I reinstalled the app, and all zim files were detected.
